govec/ts_log.go

```diff
@@ -0,0 +1,71 @@
+package govec
+
+import (
+	"bytes"
+	"github.com/DistributedClocks/GoVector/govec/vclock"
+	"os"
+	"strconv"
+	"time"
+)
+
+type GoTSLog struct {
+	GoLog
+}
+
+func InitGoVectorTimeStamp(processid string, logfilename string) *GoTSLog {
+	gv := InitGoVector(processid, logfilename)
+	gvts := &GoTSLog{*gv}
+	gvts.SetLogFunc(gvts.LogThis)
+	// Need to do a reinitialization to have a timestamped initializaiton event
+	gvts.reinitialize()
+	return gvts
+}
+
+func (gv *GoTSLog) truncate() bool {
+	complete := true
+	file, err := os.OpenFile(gv.logfile, os.O_TRUNC | os.O_WRONLY, 0600)
+	if err != nil {
+		complete = false
+	}
+	defer file.Close()
+	return complete
+}
+
+func (gv *GoTSLog) reinitialize() {
+	complete := gv.truncate()
+	if !complete {
+		gv.logger.Println("Something went wrong during re-initialization")
+	}
+	vc, err := vclock.FromBytes(gv.currentVC)
+	if err != nil {
+		gv.logger.Println(err.Error())
+	}
+	ok := gv.LogThis( "Initialization Complete", gv.pid , vc.ReturnVCString())
+	if !ok {
+		gv.logger.Println("Something went wrong during re-initialization")
+	}
+}
+
+func (gv *GoTSLog) LogThis(Message string, ProcessID string, VCString string) bool {
```

Comment:
My Go is a little shoddy, but isn't the convention that private methods start with a lower case?

Reply:
I was trying to maintain the same API as GoLog. Hence I kept it as public
Conversion of LogThis to logThis should be its own change imo.